packages/common/src/lib/common/tools.ts

```diff
@@ -3805,6 +3805,35 @@ export function compressFileGZip(
     });
 }
 
+export interface DataDirValidation {
+    /** if data directory is valid */
+    valid: boolean;
+    /** absolute path it resolves too */
+    path: string;
+    /** reason of rejection */
+    reason: string;
+}
+
+/**
+ * Validate if the dir, is a valid dataDir
+ * Data dirs in node_modules are not allowed, note that dataDirs are relative to js-controller dir or absolute
+ *
+ * @param dataDir dataDir to check
+ */
+export function validateDataDir(dataDir: string): DataDirValidation {
+    if (!path.isAbsolute(dataDir)) {
+        dataDir = path.normalize(path.join(getControllerDir(), dataDir));
+    }
+
+    const isValid = !dataDir.includes('node_modules');
```

Comment:
Playing the devils advocate here: What if I called my ioBroker folder `C:\my_node_modules\iobroker`? I guess we should split dataDir by https://nodejs.org/dist/latest-v18.x/docs/api/path.html#pathsep and check if the resulting array includes an item `node_modules`.

Reply:
Then you deserve malfunction *gg*